packages/turf-point-grid/index.js

```diff
@@ -10,60 +10,59 @@ import { getType } from '@turf/invariant';
  * @name pointGrid
  * @param {BBox|FeatureCollection|Feature} bbox extent in [minX, minY, maxX, maxY] order
  * @param {number} cellSide the distance between points
- * @param {string} [units=kilometers] used in calculating cellSide, can be degrees, radians, miles, or kilometers
- * @param {boolean} [centered=true] adjust points position to center the grid into bbox. **This parameter is going to be removed** in the next major release, having the output always centered into bbox.
- * @param {boolean} [bboxIsMask=false] if true, and bbox is a Polygon or MultiPolygon, the grid Point will be created
- * only if inside the bbox Polygon(s)
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.units="kilometers"] used in calculating cellSide, can be degrees, radians, miles, or kilometers
+ * @param {number} [options.bboxIsMask=false] if true, and bbox is a Polygon or MultiPolygon, the grid Point will be created
+ * @param {Object} [options.properties={}] passed to each point of the grid
  * @returns {FeatureCollection<Point>} grid of points
  * @example
  * var extent = [-70.823364, -33.553984, -70.473175, -33.302986];
  * var cellSide = 3;
- * var units = 'miles';
  *
- * var grid = turf.pointGrid(extent, cellSide, units);
+ * var grid = turf.pointGrid(extent, cellSide, {units: 'miles'});
  *
  * //addToMap
  * var addToMap = [grid];
  */
-export default function (bbox, cellSide, units, centered, bboxIsMask) {
```

Comment:
We need to throw an error if `options` is not an object.

```js
if (!isObject(options)) throw new Error('options is invalid')
```

Reply:
well @DenisCarriere, it can be `undefined` though

Reply:
Yes but the default value will kick in and be an object. That's why defining the default values should be before the validation, easier validation syntax